### PR TITLE
feat:added search with cookies save, test

### PR DESCRIPTION
**Pull request by Alinochka07**
Dropdown input is searchable that fetches data from API, then by clicking a specific city with address, it saves its ID as a cookie, then redirects to a new page. Redirecting is performed with dynamic route. Also, added test.

### CODE REVIEW

File: src/app/[cityEnglishName]/[id]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+type Params = {
```
Comment (Alex-2kZharkov) [inline src/app/[cityEnglishName]/[id]/page.tsx]: I think it is CitiesProps + ReactElement for return type😅

File: src/app/[cityEnglishName]/layout.tsx [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export default function CitiesLayout({ children }: { children: React.ReactNode }) {
```
Comment (Alex-2kZharkov) [inline src/app/[cityEnglishName]/layout.tsx]: PropsWithChildren from react

Comment (Alex-2kZharkov) [reply src/app/[cityEnglishName]/layout.tsx]: same for src/app/cosmetics/layout.tsx

File: src/components/DropdownCities/DropdownCities.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
+import Link from 'next/link';
+import { ReactElement } from 'react';
+import { setCookie } from 'cookies-next';
+import { CityID, SEVEN_DAYS } from '../../constants';
+import { useSetSelectedEntity } from '../../hooks/useSetSelectedEntity';
+
+export interface DropdownProps {
```
Comment (Alex-2kZharkov) [inline src/components/DropdownCities/DropdownCities.tsx]: DropdownCitiesProps

File: src/components/DropdownCities/DropdownCities.tsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+  const selectCity = (value: number) => {
+    setCookie(CityID.ID, value, { expires: SEVEN_DAYS });
+    setSelectedID(value.toLocaleString());
+  };
+
+  const handlePickCity = (value: number) => {
+    selectCity(value);
+  };
+  return (
```
Comment (Alex-2kZharkov) [inline src/components/DropdownCities/DropdownCities.tsx:38]: 1 empty line

File: src/components/DropdownInput/DropdownInput.test.tsx [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+
+import { DropdownInput } from './DropdownInput';
+
+describe('Test dropdown input component', () => {
```
Comment (Alex-2kZharkov) [inline src/components/DropdownInput/DropdownInput.test.tsx]: the description of describe block is only the name of the component you are testing, so it is `DropdownInput `

File: src/components/DropdownInput/DropdownInput.test.tsx [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+  test('Should render correctly', () => {
+    render(
+      <DropdownInput variant="fullWidth" placeholder="город, улица или метро" searchItem="" />,
+    );
+    const inputElement = screen.getByPlaceholderText('город, улица или метро');
+    expect(inputElement).toHaveValue('');
```
Comment (Alex-2kZharkov) [inline src/components/DropdownInput/DropdownInput.test.tsx:12]: would be cool to add `toBeInTheDocument` as well

File: src/components/DropdownInput/DropdownInput.tsx [hunk 7]
```diff
@@ -1,34 +1,35 @@
...
-  searchResultsText,
-  inputValue,
 }: DropdownInputProps): ReactElement => {
   const classes: Record<DropdownVariant, string> = {
     fullWidth: 'w-full',
     halfWidth: 'w-1/2',
     width75: 'w-9/12',
     width25: 'w-3/12',
   };
+  const [search, setSearch]: [string, (search: string) => void] = useState('');
```
Comment (Alex-2kZharkov) [inline src/components/DropdownInput/DropdownInput.tsx]: I am not sure if you need the `[string, (search: string) => void` type, it should be picked up autimatically by TS

File: src/constants.ts [hunk 8]
```diff
@@ -9,3 +9,7 @@ export enum Languages {
   EN = 'en',
   RU = 'ru',
 }
+
+export enum CityID {
```
Comment (Alex-2kZharkov) [inline src/constants.ts]: You use it for cookies, but I thought we use the `Cookies` enum from above

File: src/hooks/useSetSelectedEntity.tsx [hunk 9]
```diff
@@ -4,17 +4,21 @@ import { useState, useEffect, SetStateAction, Dispatch } from 'react';
 export const useSetSelectedEntity = (
   entityValue: string,
   cookieName: string,
-): { selectedEntity: string; setSelectedEntity: Dispatch<SetStateAction<string>> } => {
+): {
```
Comment (Alex-2kZharkov) [inline src/hooks/useSetSelectedEntity.tsx:7]: let's move this return type of the hook to src/types folder, it looks ugly when long types are defined directly in functions or classes

**Review by Alex-2kZharkov — CHANGES_REQUESTED** [review]

File: src/app/[cityEnglishName]/[id]/page.tsx (hunk 1 above)
Comment (Alinochka07) [reply src/app/[cityEnglishName]/[id]/page.tsx]: Pardon ))

**Review by Alex-2kZharkov — APPROVED** [review]

**Pull request merged**